### PR TITLE
Added Mo's Algorithm in java language

### PR DESCRIPTION
**Pull request by Anshuvats**
This algorithm has many application like sqrt decompostion algorithm.

### CODE REVIEW

**Comment by Anshuvats** [comment]
What happens,sir? This is PR for Mo's Algorithm not Bucket sort algorithm ,sir.

**Comment by Anshuvats** [comment]
If any mistake from me, what will i do?,sir

**Comment by Anshuvats** [comment]
Can you help me,sir?

**Comment by GOVINDDIXIT** [comment]
@Anshuvats No problem, Just remove the bucket sort algo from this PR.

**Comment by Anshuvats** [comment]
How can remove the bucket sort algo from this PR

**Comment by Anshuvats** [comment]
*I

**Comment by Anshuvats** [comment]
Can you Help me ,please?

**Comment by GOVINDDIXIT** [comment]
Go to your fork repo and delete the Bucket_sort.java file from there

**Comment by Anshuvats** [comment]
okay sir, I will do as soon as possible. Thanks

**Comment by Anshuvats** [comment]
@GOVINDDIXIT    then after what will i do?,sir

**Comment by GOVINDDIXIT** [comment]
Squash the commits

**Comment by Anshuvats** [comment]
okay,sir

**Comment by Anshuvats** [comment]
sir, i am not able to squash the commits.

**Comment by Anshuvats** [comment]
please,sir can you help me. i have tried many times. i am not able to do.

File: Graham_scan_Algorithm/main.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*The convex hull is the minimum closed area which can cover all given data points.
```
Comment (GOVINDDIXIT) [inline Graham_scan_Algorithm/main.cpp]: @Anshuvats Since the PR is for MO's Algorithm, Remove this file

Comment (GOVINDDIXIT) [reply Graham_scan_Algorithm/main.cpp]: To squash commit see[ this](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)

Comment (Anshuvats) [reply Graham_scan_Algorithm/main.cpp]: okay sir,i will try once again as soon as possible

**Comment by GOVINDDIXIT** [comment]
@Anshuvats I am merging this but please learn how to squash commits ASAP.

**Comment by Anshuvats** [comment]
Thank you, sir i will definitely learn.

**Pull request merged**